envi-tools/src/CSVIO.cpp

```diff
@@ -0,0 +1,69 @@
+#include "envitools/CSVIO.hpp"
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <boost/algorithm/string.hpp>
+
+using namespace envitools;
+namespace fs = boost::filesystem;
+
+std::vector<cv::Vec2i> CSVIO::ReadCSV(fs::path path)
+{
+    std::ifstream ifs(path.string());
+    if (!ifs.good()) {
+        throw std::runtime_error("File failed to open");
+    }
+    std::vector<cv::Vec2i> output;
+    std::string line;
+    std::vector<std::string> strs;
+    while (std::getline(ifs, line)) {
+        // Parse the line
+        boost::trim(line);
+        boost::split(strs, line, boost::is_any_of(","));
+        std::for_each(std::begin(strs), std::end(strs), [](std::string& t) {
+            boost::trim(t);
+        });
+        output.emplace_back(std::stoi(strs[0]), std::stoi(strs[1]));
+    }
+
+    return output;
+}
+
+void CSVIO::WriteCSV(
+    fs::path path,
+    const std::map<std::string, std::vector<double>>& res,
+    const std::vector<std::string>& header)
+{
+    std::ofstream ofs;
+    ofs.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+    try {
+        ofs.open(path.string());
+
+        if (!(header.empty())) {
```

Comment:
This entire thing can just be a range-based for loop:
```
for(auto h : header) {
  ofs << h << ",";
}
ofs << std::endl;
```
If header is empty, nothing will be written.